python-backend/app/api/mines/variances/resources/variance.py

```diff
@@ -0,0 +1,75 @@
+from datetime import datetime
+from flask_restplus import Resource, reqparse
+from sqlalchemy.exc import DBAPIError
+from werkzeug.exceptions import BadRequest, NotFound
+
+from ..models.variance import Variance
+from app.extensions import api
+from ....utils.access_decorators import requires_role_mine_view, requires_role_mine_create
+from ....utils.resources_mixins import UserMixin, ErrorMixin
+
+
+class VarianceResource(Resource, UserMixin, ErrorMixin):
+    parser = reqparse.RequestParser()
+    parser.add_argument('compliance_article_id',
+                        type=int,
+                        help='ID representing the MA or HSRCM code to which this variance relates.')
+    parser.add_argument('note',
+                        type=str,
+                        help='A note to include on the variance. Limited to 300 characters.')
+    parser.add_argument('issue_date',
+                        type=lambda x: datetime.strptime(x, '%Y-%m-%d') if x else None,
+                        help='The date on which the variance was issued.')
+    parser.add_argument('received_date',
+                        type=lambda x: datetime.strptime(x, '%Y-%m-%d') if x else None,
+                        help='The date on which the variance was received.')
+    parser.add_argument('expiry_date',
+                        type=lambda x: datetime.strptime(x, '%Y-%m-%d') if x else None,
+                        help='The date on which the variance expires.')
+
+
+    @api.doc(params={})
+    @requires_role_mine_view
```

Comment:
Please update python-backend\tests\auth\test_expected_auth.py with the new endpoints.

Reply:
Sure thing. I've added them